### DISCUSSION TITLE
mmrm for longitudinal field experiment in the domain of energy and water consumption

### DISCUSSION THREAD

**Discussion by vincentvincevin** [openpharma/mmrm#355] [Q&A] [answered]
Dear mmrm community,

I am a PhD researcher (psychologist) conducting longitudinal field experiments in the domain of energy and water consumption using digital behavioral interventions. Thus, my application area would go beyond clinical trials, however, the experimental setup is quite similar. During my search to adequately model and analyze my data, I saw the best fit to the mmrm package. Now I would like to ask the mmrm community, whether you see fit given the setup of my experiment. This would be of great benefit as I would like to prerigster my experiment.

In my experiment, I will have a daily baseline measurement of 4 weeks on the variables water, heat and electricity consumption. Afterward, I will randomly allocate around 400 participants (living in similar apartments with smart meters) to 4 conditions and do a randomization check (I expect no significant difference between the dependent and control variables). Then the experiment follows for 12 weeks. Participants in the 3 experimental groups are using a mobile application with different behavioral interventions and the goal is to investigate their effectiveness compared to the control.

As of today, I thought to model it as follows: outcome (water / heat / electricity consumption), fixed effects (1. condition, 2. time in days, 3. interaction between condition and time, 4. engagement interactions with the app, 5. interaction between engagement and condition (as it is expected that engagement might be higher in some conditions), 6. presence in the apartment on a given day (if consumption on a given day falls below a cut-off value), 7. control variable gender, 8. control variable age), random effect (at individual or group level).

Now turning to my questions:

1. Energy and water consumption data is usually distributed in a gamma-ish distribution (right-skewed) - most people consume some energy and water, however, there are few people that consume more extremely. Would the modeling with mmrm be robust against this non-normal distribution?
2. Compared to clinical trails, I would have many time points for the 120 days of experiments. Including them all as fixed effects would eat up a lot of degrees of freedom. Would it be better to distill the days to weeks or include the time variable as a random effect? 
3. Compared to clinical trails, the intervention is mediated by the use of the mobile application - would it be best to include the daily interactions as a fixed or a random effect?
4. Which rational is there to follow for clustering the random effect at individual or condition level?  

Thanks for your efforts in creating the mmrm package and looking forward to hearing from you!

Best wishes,
Vincent

**Comment by danielinteractive** [comment, marked as answer]
Thanks @vincentvincevin for reaching out!

To your questions specifically:

1. The `mmrm` package assumes normally distributed errors and thus observations. However you could try to log-transform your observations to try bringing it to the real line and also deal with the skewness. However, with `glmmTMB` you could directly use a gamma response distribution, see https://cran.r-project.org/web/packages/glmmTMB/index.html
2. Probably it would make sense to assume some regularity in the 120 days, e.g. weeks + day of week effects are often useful. That would reduce the number of parameters substantially. I don't think 120 random effects will tell you much.
3. I would say if those daily interactions are more nuisance than of interest, then random effects could make sense
4. The question is where you want to borrow information from - only between days from same individual? or from other individuals with same condition? That will guide you to the choice of clustering of the random effects.

Just would like to highlight that `mmrm` explicitly does not allow for random effects. We only have here fixed effects, with a structured covariance matrix for observations from the same cluster. In clinical trials this is patients are the cluster, and they have multiple time points. `glmmTMB` on the other hand can have both fixed as well as random effects. Note that `mmrm` addresses specific use cases in clinical trials where things like adjusted degrees of freedom, robust sandwich estimators etc. are very important and these are the main differentiators vs. `glmmTMB` for normal observations.
